corehq/apps/accounting/models.py

```diff
@@ -1243,6 +1244,13 @@ def change_plan(self, new_plan_version, date_end=None,
         assert self.is_active
         assert date_end is None or date_end >= today
 
+        if new_plan_version.subscription_set.count() >= new_plan_version.plan.max_domains:
```

Comment:
What happens if ```new_plan_version.plan.max_domains``` is ```None```?  Does this function work with non-customer plans?

Reply:
```new_plan_version.subscription_set()``` includes non-active subscriptions, and doesn't include subscriptions under the same plan but a different plan version.

Reply:
It looks like a lot of tests are getting tripped up here

Reply:
Yeah you're definitely right and I need a check for `None` here when `max_domains` isn't set.

Reply:
I didn't realize `new_plan_version.subscription_set()` included non-active subscriptions, so thanks for pointing that out.

When we talked about how to set subscriptions and plans yesterday, I though that subscriptions were related to each other through a `plan_version`. Is it actually the `plan` itself?